### PR TITLE
feat!: Update logic for SuperTokens.init and accepted parameters

### PR DESCRIPTION
**Pull request by nkshah2**
WIP

### CODE REVIEW

File: Example/app/src/test/java/com/example/example/android/ConfigTests.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
+/*
```
Comment (rishabhpoddar) [inline Example/app/src/test/java/com/example/example/android/ConfigTests.java:1]: Why is this in android folder

File: app/src/main/java/io/supertokens/session/SuperTokens.java [hunk 2]
```diff
@@ -25,39 +25,39 @@
...
 
     @SuppressWarnings("unused")
-    public static void init(Context applicationContext, @NonNull String refreshTokenEndpoint, @Nullable Integer sessionExpiryStatusCode,
-                            @Nullable Map<String, String> refreshAPICustomHeaders) throws MalformedURLException {
+    public static void init(
+            Context applicationContext,
+            @NonNull String apiDomain,
+            @Nullable String apiBasePath,
+            @Nullable Integer sessionExpiredStatusCode,
+            @Nullable String cookieDomain
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/SuperTokens.java:45]: Do we need `autoAddCredentials`?

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 3]
```diff
@@ -39,4 +43,99 @@ enum UnauthorisedStatus {
...
+                    trimmedSessionScope = trimmedSessionScope.substring(1);
+                }
+
+                return trimmedSessionScope;
+            } catch (Exception e) {
+                throw new MalformedURLException("Please provide a valid sessionScope");
+            }
+        }
+
+        public static String normaliseSessionScopeOrThrowError(String sessionScope) throws MalformedURLException {
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:101]: Remove this func

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 4]
```diff
@@ -39,4 +43,99 @@ enum UnauthorisedStatus {
...
+
+        public static String normaliseURLDomainOrThrowError(String input) throws MalformedURLException {
+            return new NormalisedURLDomain(input).getAsStringDangerous();
+        }
+
+        public static String normaliseURLPathOrThrowError(String input) throws MalformedURLException {
+            return new NormalisedURLPath(input).getAsStringDangerous();
+        }
+
+        static String sessionScopeHelper(String sessionScope) throws MalformedURLException {
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:74]: Remove this func

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 5]
```diff
@@ -39,4 +43,99 @@ enum UnauthorisedStatus {
...
+            this.cookieDomain = cookieDomain;
+        }
+
+        public static String normaliseURLDomainOrThrowError(String input) throws MalformedURLException {
+            return new NormalisedURLDomain(input).getAsStringDangerous();
+        }
+
+        public static String normaliseURLPathOrThrowError(String input) throws MalformedURLException {
+            return new NormalisedURLPath(input).getAsStringDangerous();
+        }
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:72]: remove this too

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 6]
```diff
@@ -39,4 +43,99 @@ enum UnauthorisedStatus {
...
+
+            int _sessionExpiredStatusCode = 401;
+            if (sessionExpiredStatusCode != null) {
+                _sessionExpiredStatusCode = sessionExpiredStatusCode;
+            }
+
+            String _cookieDomain = null;
+            if (cookieDomain != null) {
+                _cookieDomain = normaliseSessionScopeOrThrowError(cookieDomain);
+            }
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:136]: Not required

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 7]
```diff
@@ -39,4 +43,99 @@ enum UnauthorisedStatus {
             this.error = error;
         }
     }
+
+    public static class NormalisedInputType {
+        String apiDomain;
+        String apiBasePath;
+        int sessionExpiredStatusCode;
+        String cookieDomain;
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:51]: Not required to have cookieDomain

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 8]
```diff
@@ -48,94 +50,42 @@ public static class NormalisedInputType {
...
-            if (sessionScope.startsWith(".")) {
-                return "." + noDotNormalised;
-            }
-
-            return noDotNormalised;
-        }
+//        @TestOnly
+//        public static String normaliseURLPathOrThrowError(String input) throws MalformedURLException {
+//            return new NormalisedURLPath(input).getAsStringDangerous();
+//        }
```
Comment (rishabhpoddar) [inline app/src/main/java/io/supertokens/session/Utils.java:68]: Remove this comment

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/supertokens/session/Utils.java [hunk 9]
```diff
@@ -48,94 +50,42 @@ public static class NormalisedInputType {
         String apiDomain;
         String apiBasePath;
         int sessionExpiredStatusCode;
-        String cookieDomain;
 
         // TODO NEMI: Handle pre API and on handle event
         public NormalisedInputType(
                 String apiDomain,
                 String apiBasePath,
```
[the remark this answers is not in the document]
Comment (nkshah2) [reply app/src/main/java/io/supertokens/session/Utils.java:57]: `NormalisedInputType` in supertokens-website has them stored as strings

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**